src/GafferScene/Shader.cpp

```diff
@@ -150,14 +150,14 @@ class Shader::NetworkBuilder
 
 	public :
 
-		NetworkBuilder( const Shader *rootNode )
-			:	m_rootNode( rootNode ), m_handleCount( 0 )
+		NetworkBuilder( const Gaffer::Plug *output )
```

Comment:
I _think_ it's alright still. Unless the whole `correspondingInput()` thing comes into play, the generated network is the same regardless of which output plug is passed to the NetworkBuilder constructor, right? In which case the hash should be the same for each case.

Then when the node is disabled and `correspondingInput()` does come into play, you're using `effectiveParameter()` to skip over the node to the appropriate prior node, and then we're returning the hash from that.

Are you seeing problems? Either way maybe we should add a test case for some contrived shader that has two outputs each with a different corresponding input, and then assert that the hashes and generated networks are different when that shader is disabled?